Auto init test helpers (#168)
toomuchdesign/next-page-tester: README.md

| **dotenvFile**                 | Relative path to a `.env` file holding [environment variables][next-docs-env-vars] | `string`                          | -               |
| **wrapper**                    | Map of render functions. Useful to decorate component tree with mocked providers.  | `{ Page?: NextPage => NextPage }` | -               |

## Skipping Auto Cleanup & Helpers Initialisation

Since Next.js is not designed to run in a JSDOM environment we need to **setup the default JSDOM** to allow a smoother testing experience. By default, `next-page-tester` will:

> - Provide a `window.scrollTo` mock
> - Provide a `IntersectionObserver` mock

Reviewer: ```suggestion
- Provide `window.scrollTo` and `IntersectionObserver` mocks
```

- Cleanup DOM after each test

> - Isolate some React modules to preserve their identity between "server" and "client" execution

Reviewer: ```suggestion
- Setup jest to preserve the identity of some specific modules between "server" and "client" execution
```

However, you may choose to skip the auto cleanup & helpers initialisation by setting the NEXT_PAGE_TESTER_SKIP_AUTO_INIT_TEST_HELPERS env variable to 'true'. You can do this with cross-env like so:

```js

> cross-env NEXT_PAGE_TESTER_SKIP_AUTO_INIT_TEST_HELPERS=true jest

Reviewer: What about?

- `NPT_SKIP_AUTO_SETUP`
- `NEXT_PAGE_TESTER_SKIP_AUTO_SETUP`
- `NPT_SKIP_AUTO_INITIALIZATION`
- `NEXT_PAGE_TESTER_SKIP_AUTO_INITIALIZATION`
- ...?
